ddtrace/internal/telemetry/writer.py

```diff
@@ -508,6 +511,33 @@ def _app_integrations_changed_event(self, integrations):
         }
         self.add_event(payload, "app-integrations-change")
 
+    def _app_client_configuration_changed_event(self, configuration):
+        # type: (List[Dict]) -> None
+        """Adds a Telemetry event which sends list of modified configurations to the agent"""
+
+        config_change_queue = []
+        # Loop though the events queue and find the app-started event payload
+        for e in self._events_queue:
+            if e["request_type"] == "app-started":
```

Comment:
app-client-configuration-changed should not update `app-started`. It should generate another event. 

You should a pattern similar to app-integrations-changed event:
  1) Create a list to store configurations (you've done that: `_configuration_queue`) ✅ 
  2) Create a method that can flush the integrations queue (ex: https://github.com/DataDog/dd-trace-py/blob/1.x/ddtrace/internal/telemetry/writer.py#L530)
  3) Add a method that takes the queued configurations and generates an [app-client-configurations-changed payload](https://github.com/DataDog/instrumentation-telemetry-api-docs/blob/main/GeneratedDocumentation/ApiDocs/v2/SchemaDocumentation/Schemas/app_client_configuration_change.md#app_client_configuration_change) (ex: https://github.com/DataDog/dd-trace-py/blob/1.x/ddtrace/internal/telemetry/writer.py#L503)
  4) Ensure the method from steps 2 and 3 are used in TelemetryWriter.periodic (ex: 
  5) Add method that can be used to add configurations, TelemetryWriter.add_configuration(...) (ex: https://github.com/DataDog/dd-trace-py/blob/v1.15.0/ddtrace/internal/telemetry/writer.py#L499-L501)
  6) Write tests to assert that the `app-client-configurations-changed ` is sent when `TelemetryWriter.add_configuration(...)` is used

Reply:
`app-client-configurations-change` should have an almost identical implementation to `app-integrations-change`.

Feel free to message me if the steps above are not clear. Also the implementation above is just a suggestions. You can try out different approaches.